app/indexers/curator/collection_indexer.rb

```diff
@@ -20,10 +20,19 @@ class CollectionIndexer < Curator::Indexer
         accumulator << 'Collections'
         # iterate over child DigitalObject and get genre values
         # TODO: find a better way? (query Solr?), this is probably pretty expensive
-        Curator.digital_object_class.with_metastreams.where(admin_set_id: record.id).find_each do |obj|
-          obj.descriptive.genres.basic_genres.find_each do |genre|
-            accumulator << genre.label unless accumulator.include?(genre.label)
-          end
+        # UPDATE I Think I found the better way. Please Review on PR - Ben
```

Comment:
Yeah, this is a bit faster (tested with Benchmark a few times) -- go ahead and remove the previous code (currently commented out).